### PR TITLE
CI: Change Required Check list, few fixes

### PR DESCRIPTION
**Pull request by maxknv**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
CI fixes

- Add Stateful asan test, Stateless asan, Stateless flaky chek into Required
- Move binary_release to normal builds (not special). It builds fast and there are test jobs depending on it.
- Add job description for A Sync

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

<details>
    <summary>CI Settings</summary>

**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step
- [ ]  Allow: Integration Tests
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Unit tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All with aarch64
- [ ]  Allow: All with ASAN
- [ ]  Allow: All with TSAN
- [ ]  Allow: All with Analyzer
- [ ]  Allow: All with Azure
- [ ]  Allow: Add your option here
---
- [ ]  Exclude: Fast test
- [ ]  Exclude: Integration Tests
- [ ]  Exclude: Stateless tests
- [ ]  Exclude: Stateful tests
- [ ]  Exclude: Performance tests
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN
- [ ]  Exclude: All with MSAN
- [ ]  Exclude: All with UBSAN
- [ ]  Exclude: All with Coverage
- [ ]  Exclude: All with Aarch64
---
- [x]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)
- [ ]  allow: batch 1 for multi-batch jobs
- [ ]  allow: batch 2
- [ ]  allow: batch 3
- [ ]  allow: batch 4, 5 and 6
</details>

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 1e3d2973462db32e74ff64352bbc20ac8c1fe7ae with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/64605/1e3d2973462db32e74ff64352bbc20ac8c1fe7ae/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>A Sync</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Comment by maxknv** [comment]
@Algunenano please check updated Required check list

File: tests/ci/ci_config.py [hunk 1]
```diff
@@ -1421,8 +1424,8 @@ def __hash__(self) -> int:
 CHECK_DESCRIPTIONS = [
     CheckDescription(
         StatusNames.SYNC,
-        "A status of the workflow running in the cloud repository against the sync PR",
-        lambda x: x == StatusNames.SYNC,
+        "If it fails, ask a maintainer for help",
+        lambda x: "Sync" in x,
```
Comment (Felixoid) [inline tests/ci/ci_config.py:1428]: No, why? It must be the exact name.

Any example where this `in` is necessary?

**Review by Felixoid — CHANGES_REQUESTED** [review]
In the sync PR we should add a validation for `all(job in REQUIRED_CHECKS for job in SYNC_JOB_LIST)`

We can do it right in the body of ci_config.py.

File: tests/ci/ci_config.py (hunk 1 above)
Comment (maxknv) [reply tests/ci/ci_config.py:1428]: it's a minor thing, i didn't know you change it too

Comment (Felixoid) [reply tests/ci/ci_config.py]: The description is subjective; this one is good, too.

But we shouldn't have other `Sync` statuses, hopefully.

**Review by Felixoid — APPROVED** [review]
Let's ensure no `Tests_3` will be possible in the sync, and it's good to go. Thanks!@

**Pull request merged**